source/game.database.localizer.bmx

```diff
@@ -25,11 +25,12 @@ Type TDatabaseLocalizer
 
 	Method getGlobalVariable:String(languageCode:String, key:String, fallback:Int=True)
 		Local l:TLocalizationLanguage = getGlobalVariables(languageCode)
-		If l And l.Has(key) Then Return l.Get(key)
+		Local lowerKey:String = key.toLower()
```

Comment:
An anderen Stellen hatte ich das bereits eingefuehrt ... ein bool-Parameter "IsLowerKey:Int=False" oder aehnliches.

Sinn waere hier, das "toLower" nur auszufuehren, wenn wirklich noetig (damit nicht immer ein neuer String angelegt wird).

Wollen wir sowas hier auch mit einbinden? Alternativ kann ich eine Funktion bereitstellen, die zumindest fuer "unkritische Zeichen" ein "StringHelper.IsLower:int(s:String)" oder so bereitstellt ...

Reply:
Ich werde das isLowerKey noch einführen.

Reply:
An dieser Stelle mal: Ich hatte das mit dem string.ToLower() mal gegenueber Brucey erwaehnt - ich wollte flags (im Stringtyp) einfuehren, damit es nur einmalig checkt ob "is lower" (und dann den Flag speichert).

Er hatte dann mal den Code durchgeschaut (haette ich wohl auch machen koennen) ... wenn "mystring.ToLower()" den gleichen String wie "mystring" zurueckgeben wuerde, wuerde kein neuer String erzeugt werden, sondern der alte benutzt.

An manchen Stellen kostet es also nur "CPU", wenn bereits "lower strings" uebergeben werden (man sich halt nur nicht sicher ist...)